### ISSUE TITLE
Differential clauses are not taken into account when validating against a custom structure definition

### ISSUE THREAD

**Issue by aberge** [jamesagnew/hapi-fhir#816]
I am trying to validate my resources against a custom structure definition for DSTU3. Although constraints on cardinality, fixedCode... are added via the differential elements, the validator does not report any error related to those constraints.

StructureDefinition snippet is provided below.
```
  <baseDefinition value="http://ihe.net/fhir/StructureDefinition/IHE.mCSD.Location" />
  <derivation value="constraint" />
  <differential>
    <element id="Location.position">
      <path value="Location.position" />
      <min value="1" />
    </element>
  </differential>
```
I do reference this structure definition when creating the validator:

  ```
  PrePopulatedValidationSupport valSupport = new PrePopulatedValidationSupport();
    DefaultProfileValidationSupport defaultSupport = new DefaultProfileValidationSupport();
    ValidationSupportChain support = new ValidationSupportChain(valSupport, defaultSupport);
    valSupport.addStructureDefinition(structureDefinition);
    FhirContext context = FhirContextProvider.instance();
    FhirValidator customStructureDefValidator = context.newValidator();
    FhirInstanceValidator fhirInstanceValidator = new FhirInstanceValidator();
    fhirInstanceValidator.setValidationSupport(support);
    customStructureDefValidator.registerValidatorModule(fhirInstanceValidator);
```
Anyway, when I validate the resource below:
```
<Location xmlns="http://hl7.org/fhir">
    <id value="007"/>
    <text>
        <status value="generated"/>
    </text>
</Location>
```
The validator only reports the following:
  ```
<Severity>error</Severity>
            <Message>Location.text: Profile http://hl7.org/fhir/StructureDefinition/Narrative, Element 'Location.text.div': minimum required = 1, but only found 0</Message>
            <columnNumber>0</columnNumber>
            <lineNumber>0</lineNumber>
```

I would expect that the validator complains about the missing position element.

Is there a bug or did I miss something when configuring the validator ?

**Comment by aberge** [comment]
Note that I am using the latest available version for DSTU3: 3.0.0

**Comment by jamesagnew** [comment]
Hi @aberge - The validator has been upgraded to use the R4 validator even for DSTU3 validation, so it is much more full featured.

Can you verify if this works in the latest 3.3.0-SNAPSHOT builds?
